Assets/Scripts/Models/Utility.cs

```diff
@@ -0,0 +1,900 @@
+#region License
+// ====================================================
+// Project Porcupine Copyright(C) 2016 Team Porcupine
+// This program comes with ABSOLUTELY NO WARRANTY; This is free software, 
+// and you are welcome to redistribute it under certain conditions; See 
+// file LICENSE, which is part of this source code package, for details.
+// ====================================================
+#endregion
+using System;
+using System.Collections.Generic;
+using System.Xml;
+using System.Xml.Schema;
+using System.Xml.Serialization;
+using Animation;
+using MoonSharp.Interpreter;
+using MoonSharp.Interpreter.Interop;
+using ProjectPorcupine.PowerNetwork;
+using UnityEngine;
+
+/// <summary>
+/// InstalledObjects are things like walls, doors, and furniture (e.g. a sofa).
+/// </summary>
+[MoonSharpUserData]
+public class Utility : IXmlSerializable, ISelectable, IContextActionProvider, IBuildable
```

Comment:
I'll say pretty confidently that it probably shouldn't just straight up be a furniture, while it is currently mostly a more simplified furniture, there are a few things that are more complex, and a fair bit of what furniture does it will probably never need. Buildable would likely be a reasonable base class, the only real reason I chose to go with Interface was because that's how several similar things are currently implemented, and it allowed me to minimize changes to furniture in a PR that's not directly related to it, to minimize the number of conflicts this PR has or causes.